test/integration/GuardedStream.test.ts

```diff
@@ -0,0 +1,54 @@
+import {
+  RepresentationMetadata,
+  TypedRepresentationConverter,
+  readableToString,
+  ChainedConverter,
+  CONTENT_TYPE,
+  guardedStreamFrom,
+  RdfToQuadConverter,
+} from '../../src';
+import type {
+  Representation,
+  RepresentationConverterArgs,
+} from '../../src';
+
+class DummyConverter extends TypedRepresentationConverter {
+  public async getInputTypes(): Promise<Record<string, number>> {
+    return { '*/*': 1 };
+  }
+
+  public async getOutputTypes(): Promise<Record<string, number>> {
+    return { 'custom/type': 1 };
+  }
+
+  public async handle({ representation }: RepresentationConverterArgs): Promise<Representation> {
+    const data = guardedStreamFrom([ 'dummy' ]);
+    const metadata = new RepresentationMetadata(representation.metadata);
+
+    return { binary: true, data, metadata };
+  }
+}
+
+describe('A chained converter where data gets ignored', (): void => {
+  const identifier = { path: 'http://test.com/' };
+  const metadata = new RepresentationMetadata(identifier, { [CONTENT_TYPE]: 'text/turtle' });
+  const rep = {
+    binary: true,
+    data: guardedStreamFrom([ '<a:b> <a:b> c!' ]),
+    metadata,
+  };
+  const converter = new ChainedConverter([
+    new RdfToQuadConverter(),
+    new DummyConverter(),
+  ]);
+
+  it('does not throw an async crash.', async(): Promise<void> => {
+    jest.useFakeTimers();
+    const result = await converter.handleSafe({ identifier, representation: rep, preferences: {}});
+
+    expect(await readableToString(result.data)).toBe('dummy');
+
+    // Prevent logging after all the tests are finished
+    jest.advanceTimersByTime(1000);
```

Comment:
Or we can mock the logger (and maybe we should do that in tests actually). https://github.com/solid/community-server/issues/499

Reply:
The problem is the the logger is not part of an object and is instantiated in the util file itself and I actually haven't figured out a good way to mock it then.

Reply:
Good point, never mind.

Reply:
Funny story: apparently I did figure out how to mock that logger since that is exactly what happens in the guarded stream unit tests, so I can just copy that here.

Reply:
You're welcome https://github.com/solid/community-server/blame/b32da0de1a9f1edc84810151508143d5b9a1b54b/test/unit/util/GuardedStream.test.ts

Reply:
This explains a lot of my confusion while writing this PR